### ISSUE TITLE
Implement (Non-central) Distributions in Rust

### ISSUE THREAD

**Issue by storopoli** [poweranalyses-org/poweranalyses#24]
If we implement the (non-central) distributions in Rust, we can drop the `nmath` C library and `emscripten` as dependencies.

What we need:

- Distributions with `cdf` (and hence the `pdf` is included) and `quantile` functions:
  - Non-central $\chi^2$
  - Non-central $F$
  - Non-central $t$
  - $z$ (Standard Normal)
  - Binomial (for some exact tests)

Alternatives:

- ~~Make a PR to [`statsrs`](https://github.com/statrs-dev/statrs)~~ it is dead.
- Implement our own in a `lib` sub-crate.
  - The code can be ported from Julia's [`Distributions.jl`](https://github.com/JuliaStats/Distributions.jl/tree/master/src/univariate/continuous)
  - The math and formulas can be verified at [Wikipedia](https://en.wikipedia.org/wiki/Noncentral_distribution)

**Comment by storopoli** [comment]
> Dropping emscripten would be great because then we could move `frontend.js` into the back end 🥳 via `web-sys`

Does this means that we can strongly type the API inputs and outputs and get ride of `parse_[i|f]64`?

**Comment by rikhuijzer** [comment]
> Does this means that we can strongly type the API inputs and outputs and get rid of `parse_[i|f]64`?

No, what would change is that instead of interacting with the DOM using Javascript (for example, via `getElementById(id)`), we could interact directly with the DOM from within Rust. This would mean that we could drop the JSON communication between back end and front end because the back end can directly modify the DOM elements.

**Comment by rikhuijzer** [comment]
> Ok so we have `'s/JavaScript/Rust'` in all frontend? But if we drop the JSON communication aren't we also dropping the parsing floats and ints into straight getting Rust's floats and ints?

Could be. I don't know whether HTML number elements are automatically interpreted as number types by `web-sys`.
